Breaking news, ProcessTest is failing intermttently

This is a note for the future and for folks who monitor CI Integration failures.
Sorry, but I have absolutely zero time to chase this now. Perhaps in
a kinder future.


The CI run after PR #2206 had an intermittent failure in ProcessTest:
```

Error:  Test java.lang.ProcessTest.outputStreamReadsFromFile failed: java.lang.AssertionError: Process took more than 10 SECONDS to exit., took 10.023 sec
```

I believe that, as annoying as intermittent failures are, the Test is doing its job and SN development
a service.    The 10 second value for the timeout was (several) order of magnitude of what a
reasonable process exit time would be, even on a heavily burdened system.

The fact that it is taking anywhere near ten seconds to  for the Process to read two
lines of input and then exit is almost a certain indicator of bogosity.

I took a quick look at the echo.sh in question and consulted the Posix sh specification.
echo.sh is suspect (why the || true??? my imagination fails me).  

The buffering on the FileOutputStream in the ProcessTest test could probably
be improved by closing the file (I believe that FileOutputStream#flush does
nothing.)  One would expect the line terminating with newlines to cause
any underlying buffering to flush, but....

For an debugging trail...

Usually succeeds but  now seen a second time
```
2021-05-15T05:02:25.1579713Z [info] Test java.lang.ProcessTest.outputStreamReadsFromFile started
2021-05-15T05:02:25.1581545Z [error] Test java.lang.ProcessTest.outputStreamReadsFromFile failed: java.lang.AssertionError: Process took more than 10 SECONDS to exit., took 10.009 sec
2021-05-15T05:02:25.1583434Z [error]     at java.lang.StackTrace$.currentStackTrace(Unknown Source)
2021-05-15T05:02:25.1584575Z [error]     at java.lang.Throwable.fillInStackTrace(Unknown Source)
```
There is no way that correct code should be taking 10 seconds.  GitHub CI might be swapping 
out or such.  The timeout is _just_ on the borderline.      

Lee: Trace this when you can. Close the file to force a synch?  Should not have to.

--- a/unit-tests/src/test/scala/java/lang/ProcessTest.scala
+++ b/unit-tests/src/test/scala/java/lang/ProcessTest.scala
@@ -134,11 +134,11 @@ class ProcessTest {
     pb.redirectInput(file)
 
     try {
-      val proc = pb.start()
       val os = new FileOutputStream(file)
       os.write("hello\n".getBytes)
       os.write("quit\n".getBytes)
 
+      val proc = pb.start()
       assertProcessExitOrTimeout(proc)
 
       assertEquals("hello", readInputStream(proc.getInputStream))

